skbio/metadata/_interval.py

```diff
@@ -0,0 +1,756 @@
+        --------
+        skbio.metadata.IntervalMetadata.drop
+        '''
+        if not self.dropped:
+            self._interval_metadata.drop([self])
+
+    def _locations_boundaries_setter(self, locations=None, boundaries=None):
+        if self.dropped:
+            raise RuntimeError('Cannot change the dropped '
+                               'Interval object.')
+        # Casts to `list`, validation, sorting, and setting of `locations`
+        # and `boundaries` happen here.
+        if locations is not None:
+            # check iterability
+            try:
+                # check iterability
+                locations = list(locations)
+            except TypeError:
+                raise TypeError(
+                    'Cannotgive an iterable (%r) to `locations`.' % locations)
+
+            # check it is not empty
+            if not locations:
+                raise ValueError('Cannot give empty `location`.')
+            # check each contiguous span is in right format
+            for location in locations:
+                _assert_valid_location(location)
+
+            spans = len(locations)
+        else:
+            spans = len(self.locations)
+
+        if boundaries is not None:
+            try:
+                boundaries = list(boundaries)
+            except TypeError:
+                raise TypeError(
+                    'Cannot give a non-iterable (%r) '
+                    'to `boundaries`.' % boundaries)
+
+            if len(boundaries) != spans:
+                raise ValueError(
+                    'The length of boundaries must '
+                    'be equal to the length of locations {!r}.')
+
+            for boundary in boundaries:
+                _assert_valid_boundary(boundary)
+
+        if locations is None:
+            # `locations` and `boundaries` cannot both be omitted.
+            if boundaries is None:
+                raise ValueError('You must give `None` to both `locations` '
+                                 'and `boundaries`.')
+            # If only `boundaries` is provided, set `self.boundaries` and don't
+            # change `self.locations`.
+            else:
+                self._boundaries = boundaries
+        else:
+            # If only `locations` is provided, reset `self.boundaries` to
+            # all `True`.
+            if boundaries is None:
+                locations.sort()
+                self._locations = locations
+                # reset all the boundaries to True!!
+                del self.boundaries
+
+            # If both `locations` and `boundaries` are provided, set
+            # `self.locations` and `self.boundaries`.
+            else:
+                combo = [i for i in sorted(zip(locations, boundaries))]
+                # use the boundaries setter to validate the value
+                self._boundaries = [i[1] for i in combo]
+                self._locations = [i[0] for i in combo]
+
+            self._interval_metadata._is_stale_tree = True
```

Comment:
Ah I missed that, thanks!

On Friday, September 2, 2016, Zech Xu <notifications@github.com> wrote:

> In skbio/metadata/_interval.py
> <https://github.com/biocore/scikit-bio/pull/1414#discussion_r77427270>:
>
> > +            # all `True`.
> > +            if boundaries is None:
> > +                locations.sort()
> > +                self._locations = locations
> > +                # reset all the boundaries to True!!
> > +                del self.boundaries
> > +
> > +            # If both `locations` and `boundaries` are provided, set
> > +            # `self.locations` and `self.boundaries`.
> > +            else:
> > +                combo = [i for i in sorted(zip(locations, boundaries))]
> > +                # use the boundaries setter to validate the value
> > +                self._boundaries = [i[1] for i in combo]
> > +                self._locations = [i[0] for i in combo]
> > +
> > +            self._interval_metadata._is_stale_tree = True
>
> This statement is inside the if scope of locations is gonna be changed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1414/files/1837b1b85f84a93c77811f3c1a425f7da63c4a69#r77427270>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABwvwEAIe7tjkkpxmfCUX5dDl31BP_GBks5qmL1ggaJpZM4JoIbD>
> .
>


-- 
Sent from my phone.